pysignalclirestapi/api.py

```diff
@@ -3,35 +3,55 @@
 import sys
 import base64
 import json
-
+from abc import ABC, abstractmethod
 from requests.models import HTTPBasicAuth
 from future.utils import raise_from
 import requests
 from .helpers import bytes_to_base64
 
 
 class SignalCliRestApiError(Exception):
-    """SignalCliRestApiError base classi."""
+    """SignalCliRestApiError base class."""
     pass
 
 
+class SignalCliRestApiAuth(ABC):
+    """SignalCliRestApiAuth base class."""
+
+    @abstractmethod
+    def getAuth():
```

Comment:
Currently all the methods are in snake case. So I think we should rename that to `get_auth` here as well.